src/bench/readblock.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <bench/data.h>
+
+#include <chainparams.h>
+#include <consensus/validation.h>
+#include <node/blockstorage.h>
+#include <streams.h>
+#include <util/system.h>
+#include <validation.h>
+
+using node::BLOCK_SERIALIZATION_HEADER_SIZE;
+using node::OpenBlockFile;
+using node::ReadBlockFromDisk;
+using node::ReadRawBlockFromDisk;
+
+static void SetupReadBlockTest()
+{
+    // Create params and set blocksdir to temporary folder (/tmp usually on linux)
+    SelectParams(CBaseChainParams::MAIN);
+    gArgs.ForceSetArg("-blocksdir", fs::temp_directory_path().string());
+
+    // Serialize block
+    CDataStream stream(benchmark::data::block413567, SER_NETWORK, PROTOCOL_VERSION);
+    CBlock block;
+    stream >> block;
+
+    // Write block at first position in <temp_directory>/blocks/blk00000.dat
+    CAutoFile fileout(OpenBlockFile(FlatFilePos(0, 0)), SER_DISK, CLIENT_VERSION);
+    assert(!fileout.IsNull());
+    const size_t block_size{GetSerializeSize(block, fileout.GetVersion())};
+    fileout << Params().MessageStart() << (uint32_t) block_size << block;
```

Comment:
```suggestion
    fileout << Params().MessageStart() << uint32_t{block_size} << block;
```

Reply:
This gives a warning about narrowing conversion.